### PR TITLE
add search component

### PR DESCRIPTION
**Pull request by HugoLiconV**

### CODE REVIEW

File: src/views/Search.js [hunk 1]
```diff
@@ -1,7 +1,151 @@
...
+  }
+});
+
+class Search extends Component {
+  state = {
+    searchQuery: ''
+  };
+
+  search = (query, type, limit, offset) => {
+    this.props.search(query, type, limit, offset);
```
Comment (linx05) [inline src/views/Search.js]: A config object would be better than separate parameters.

File: src/views/Search.js [hunk 2]
```diff
@@ -1,7 +1,151 @@
...
+
+  /**
+   * size: small, medium, large
+   */
+  getItemImage = (images, minSize, maxSize) => {
+    if (!images || images.length === 0) return noImageFound;
+    const filteredImages = images.filter(
+      ({ width }) => width > minSize && width < maxSize
+    );
+    return filteredImages.length === 0 ? images[0].url : images[0].url;
```
Comment (linx05) [inline src/views/Search.js]: Same result for both cases?

File: src/views/Search.js [hunk 3]
```diff
@@ -1,7 +1,151 @@
...
+  };
+
+  onSubmit = event => {
+    event.preventDefault();
+    const { searchQuery } = this.state;
+    if (!searchQuery) return;
+    this.search(searchQuery, ['album', 'track', 'artist']);
+  };
+
+  areArraysEmpty = arrays => {
```
Comment (linx05) [inline src/views/Search.js]: This should probably moved out to a helper function.

Comment (linx05) [reply src/views/Search.js]: Also there's no need to use an arrow function here, as it only depends on it's parameters.

File: src/views/Search.js (hunk 2 above)
Comment (HugoLiconV) [reply src/views/Search.js]: Sorry, It should be filteredImages[0].url

File: src/views/Search.js (hunk 3 above)
Comment (HugoLiconV) [reply src/views/Search.js]: Thanks! I'll change that!

File: src/views/Search.js (hunk 1 above)
Comment (HugoLiconV) [reply src/views/Search.js]: it doesn't matter that I have default params? This is how the Search Service is declared
`search = async (query, type = [], limit = 20, offset = 0) => {`

**Pull request merged**